components/eam/src/physics/cam/convect_deep.F90

```diff
@@ -1,428 +1,331 @@
+   real(r8), dimension(pcols,pver), intent(out) :: dp       ! layer thickness between interfaces   [mb]
+   real(r8), dimension(pcols),      intent(out) :: dsubcld  ! layer thickness between lcl and maxi [mb]
+   integer,  dimension(pcols),      intent(out) :: jt       ! top level index of deep convection
+   integer,  dimension(pcols),      intent(out) :: maxg     ! gathered values of maxi
+   integer,  dimension(pcols),      intent(out) :: ideep    ! flag to indicate active column for gathering
+   integer,                         intent(out) :: lengath  ! number of columns in gathered arrays
+   !----------------------------------------------------------------------------
+   ! Local variables
+   integer,           dimension(pcols) :: jctop       ! integer version of cloud top level index
+   integer,           dimension(pcols) :: jcbot       ! integer version of cloud bot level index
+   real(r8), pointer, dimension(:)     :: jctop_r8    ! real version of cloud top level index for pbuf
+   real(r8), pointer, dimension(:)     :: jcbot_r8    ! real version of cloud bot level index for pbuf
+   real(r8), pointer, dimension(:)     :: prec        ! total precip at surface
+   real(r8), pointer, dimension(:)     :: snow        ! snow at surface
+   real(r8), pointer, dimension(:,:)   :: ql          ! grid slice of cloud liquid water
+   real(r8), pointer, dimension(:,:)   :: qi          ! wg grid slice of cloud ice
+   real(r8), pointer, dimension(:,:)   :: rprd        ! rain production rate
+   real(r8), pointer, dimension(:,:)   :: evapcdp     ! evaporation of deep convective precip
+   real(r8), pointer, dimension(:)     :: pblh        ! planetary boundary layer height
+   real(r8), pointer, dimension(:)     :: tpert       ! thermal temperature excess 
+   real(r8), pointer, dimension(:,:,:) :: fracis      ! fraction of transported species that are insoluble
+   real(r8), pointer, dimension(:,:)   :: mudpcu      ! Droplet size distribution shape parameter for radiation
+   real(r8), pointer, dimension(:,:)   :: lambdadpcu  ! Droplet size distribution shape parameter for radiation
+   real(r8), pointer, dimension(:,:)   :: ttend_dp    ! temperature tendency from deep convection
+   integer :: i, k ! loop iterators
+   !----------------------------------------------------------------------------
+
+   ! Associate pointers with physics buffer fields
    call pbuf_get_field(pbuf, cldtop_idx,  jctop_r8 )
    call pbuf_get_field(pbuf, cldbot_idx,  jcbot_r8 )
    call pbuf_get_field(pbuf, icwmrdp_idx, ql    )
 
   select case ( deep_scheme )
-  case('off', 'CLUBB_SGS' ) !    0 ==> no deep convection
-    zero = 0     
-    mcon = 0
-    dlf = 0
-    pflx = 0
-    cme = 0
-    zdu = 0
-    rliq = 0
-    rice = 0   
-
-    call physics_ptend_init(ptend, state%psetcols, 'convect_deep')
-
-!
-! Associate pointers with physics buffer fields
-!
-
-    call pbuf_get_field(pbuf, cld_idx,         cld,    start=(/1,1/),   kount=(/pcols,pver/) ) 
```

Comment:
Is there a reason why this is left out in revisions? Or was it moved?

Reply:
This field wasn't being used anywhere. At some point there was a line that set it to zero, but Hui Wan commented it out with a note that it was essentially a bug. So I just removed all mention of this field to avoid confusion.